xlml/apis/task.py

```diff
@@ -100,22 +125,68 @@ def provision(self) -> Tuple[DAGNode, airflow.XComArg, airflow.XComArg]:
 
       queued_resource_op, queued_resource_name = tpu.create_queued_resource(
           tpu_name,
-          self.task_test_config.accelerator,
           self.task_gcp_config,
           ssh_keys,
           self.tpu_create_timeout,
-          self.task_test_config.num_slices,
+          self.task_test_config,
       )
-      queued_resource_op >> tpu.ssh_tpu.override(task_id="setup")(
-          queued_resource_name,
-          # TODO(wcromar): remove split
-          self.task_test_config.setup_script,
+
+      setup = self.setup(queued_resource_name, ssh_keys)
+      queued_resource_op >> setup
```

Comment:
As the start_script method is not using the setup. Can we revert to the original implementation? So we could wrap up set_up inside of provision.

Reply:
setup is still in `provision`:
https://screenshot.googleplex.com/ByYwjXZa6t9TZrF

I am just wrapping setup into a function in line 172, this looks cleaner to me, what do you think?

Reply:
Done.